drivers/ethernet/eth_stm32_hal_v2.c

```diff
@@ -279,7 +293,11 @@ int eth_stm32_tx(const struct device *dev, struct net_pkt *pkt)
 		HAL_ETH_ReleaseTxPacket(heth);
 	} else {
 		/* We need to release the tx context and its buffers */
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32mp13_ethernet)
+		HAL_ETH_TxFreeCallback(heth, (uint32_t *)ctx);
+#else
 		HAL_ETH_TxFreeCallback((uint32_t *)ctx);
+#endif
```

Comment:
Proposal:
To avoid redundant ifdefery in the function headers, you can define a custom arg:
```
		HAL_ETH_TxFreeCallback(STM32_ETH_ARGS(heth, (uint32_t *)ctx));
```

With:
```
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32mp13_ethernet)
#define STM32_ETH_ARGS(foo, VARGS)  foo, VARGS
#else
#define STM32_ETH_ARGS(foo, VARGS)  VARGS
#endif
```